source/Calamari.Azure/Scripts/DeployAzureServiceFabricApplication.ps1

```diff
@@ -124,90 +124,79 @@ $manifestXml = [Xml] (Get-Content $ManifestFilePath)
+$TypeAndVersionExists = (Get-ServiceFabricApplicationType -ApplicationTypeName $AppTypeName | Where-Object  { $_.ApplicationTypeVersion -eq $AppTypeVersion }) -ne $null
+$requiresRegister = $false
+
+Write-Verbose "App Type Name And Version Exists: $TypeAndVersionExists"
+Write-Verbose "Application Type and Name Exists: $AppTypeAndNameExists"
+Write-Verbose "Application Type and Name and Version Exists: $AppTypeAndNameAndVersionExists"
+
+Write-Verbose "Application Type Name: $AppTypeName"
+Write-Verbose "Application Type Version: $AppTypeVersion"
+Write-Verbose "Application Name: $AppName"
+Write-Verbose "Is Upgrade: $IsUpgrade"
+Write-Verbose "Found Application: '$AppTypeName' with version '$AppTypeVersion'"
+
+$parameters = @{
+    ApplicationPackagePath =  $ApplicationPackagePath
+    ApplicationParameterFilePath = $publishProfile.ApplicationParameterFile
+    ApplicationParameter = $ApplicationParameter
+    SkipPackageValidation = $SkipPackageValidation
+}
 
-    $UpgradeParameters = $publishProfile.UpgradeDeployment.Parameters
+Write-Verbose "Parameters: "
+Write-Verbose $($parameters | Out-String)
 
-    if ($OverrideUpgradeBehavior -eq 'ForceUpgrade') {
-        # Warning: Do not alter these upgrade parameters. It will create an inconsistency with Visual Studio's behavior.
-        $UpgradeParameters = @{ UnmonitoredAuto = $true; Force = $true }
-    }
-        
+if (-not $TypeAndVersionExists) {
     if ($CopyPackageTimeoutSec) {
         $parameters.CopyPackageTimeoutSec = $CopyPackageTimeoutSec
     }
 
     if ($RegisterApplicationTypeTimeoutSec) {
-        Get-Help Publish-UpgradedServiceFabricApplication -Parameter RegisterApplicationTypeTimeoutSec -ErrorVariable timeoutParamMissing -ErrorAction SilentlyContinue | Out-Null
+        Get-Help Publish-NewServiceFabricApplication -Parameter RegisterApplicationTypeTimeoutSec -ErrorVariable timeoutParamMissing -ErrorAction SilentlyContinue | Out-Null
         if (!$timeoutParamMissing) {
             $parameters.RegisterApplicationTypeTimeoutSec = $RegisterApplicationTypeTimeoutSec
         } else {
             Write-Warning "A value was supplied for RegisterApplicationTypeTimeoutSec but the current Service Fabric SDK doesn't support it."
         }
     }
 
-    Write-Verbose "Calling Publish-UpgradedServiceFabricApplication"
-    Write-Verbose "Parameters: "
-    Write-Verbose $($parameters | Out-String)
-    Write-Verbose "Upgrade parameters: "
-    Write-Verbose $($UpgradeParameters | Out-String)
-
-    Publish-UpgradedServiceFabricApplication @parameters -UpgradeParameters $UpgradeParameters -ErrorAction Stop
+    #Write-Host "Performing '$($parameters.Action)' action"
+    #Publish-UpgradeServiceFabricApplication @parameters -ErrorAction Stop
+    $requiresRegister = $true
 }
-else
+
+if ($AppTypeAndNameExists -and -not $AppTypeAndNameAndVersionExists )
 {
-    $Action = "RegisterAndCreate"
-    if ($DeployOnly) {
```

Comment:
Yes, we intentionally left some things out at this point, since we weren't entirely sure whether those things actually made sense. It's entirely possible there is a reason you may want to do register the type only, but struggling to see the use case. Happy to add it back in to avoid breaking changes though 👍